### DISCUSSION TITLE
Vercel enable cors error Mixed Routing Properties

### DISCUSSION THREAD

**Discussion by tomerdad** [vercel/community#673] [Help] [answered]
Hello, I'm trying to enable cors but it not working,
this is the `vercel.json`
```
{
  "version": 2,
  "env": {
      "DATABASE_URI": "sqlite:///tmp/demo.sqlite"
  },
  "builds": [
    {
      "src": "./index.js",
      "use": "@vercel/node",
      "config": {
        "bundle": false
      }
    }
  ],
  "routes": [
      {
          "src": "/(.*)",
          "dest": "/"
      }
  ]
}
```
and this is the `next.config.js`
```
const nextConfig = {
  reactStrictMode: true,
  env: {
    IP: 'https://jobs-server.vercel.app/',
  },
  async headers() {
    return [
      {
        // matching all API routes
        source: "https://jobs-server.vercel.app/api/*",
        headers: [
          { key: "Access-Control-Allow-Credentials", value: "true" },
          { key: "Access-Control-Allow-Origin", value: "*" },
          { key: "Access-Control-Allow-Methods", value: "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
          { key: "Access-Control-Allow-Headers", value: "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" },
        ]
      }
    ]
  }
}

module.exports = nextConfig
```
https://vercel.com/support/articles/how-to-enable-cors
according to this I need to add headers to the vercel.json file but it gives me a `Mixed Routing Properties` error 
https://vercel.com/docs/errors#errors/mixed-routing-properties
How can I do it right? Thanks

**Comment by amyegan** [comment, marked as answer]
Hi @tomerdad,

The `Mixed Routing Properties` error is being caused by the legacy `routes` property in your vercel.json config. Upgrading it to `"rewrites": [{ "source": "/(.*)", "destination": "/" }]` should solve your problem. 

More details on this are available at [https://vercel.com/docs/project-configuration#legacy/routes/upgrading](https://vercel.com/docs/project-configuration#legacy/routes/upgrading)
